File: Tinkoff_chat/Controllers/ThemesViewController/ThemesViewController.m
//
//  ThemesViewController.m
//  Tinkoff_chat
//
//  Created by Denis Nefedov on 06/03/2019.
//  Copyright © 2019 X. All rights reserved.
//

#import "ThemesViewController.h"
#import "Themes.h"
#import "ThemesViewControllerDelegate.h"


@implementation ThemesViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _model = [[Themes alloc] initColor1:UIColor.redColor
                                 color2:UIColor.blueColor
                                 color3:UIColor.greenColor];
}

-(void) viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    [self.view setBackgroundColor:[[self.navigationController navigationBar] barTintColor]];
}

- (void) setDelegate:(id<ThemesViewControllerDelegate>)delegate{
    _delegate = delegate;
}

- (id<ThemesViewControllerDelegate>) delegate {
    return _delegate;
}

- (void)choosedThemeButton:(UIButton *)sender {
    if (sender == [self.themeButtons objectAtIndex:0]) {
        [_delegate themesViewController:self didSelect:[_model theme1]];
        [[self.navigationController navigationBar] setBarTintColor:[_model theme1]];
    } else if (sender == [self.themeButtons objectAtIndex:1]) {
        [_delegate themesViewController:self didSelect:[_model theme2]];
        [[self.navigationController navigationBar] setBarTintColor:[_model theme2]];
    } else {
        [_delegate themesViewController:self didSelect:[_model theme3]];
        [[self.navigationController navigationBar] setBarTintColor:[_model theme3]];
    }
}

- (void)dealloc {
    [_themeButtons release];
    [_model release];
    [super dealloc];
}



@end

File: Tinkoff_chat/Controllers/ThemesViewController/ThemesViewController.swift
//
//  ThemesViewController.swift
//  Tinkoff_chat
//
//  Created by Denis Nefedov on 06/03/2019.
//  Copyright © 2019 X. All rights reserved.
//

import UIKit

class ThemesViewController: UIViewController {

    @IBOutlet var themeButtons: [UIButton]!

    typealias ThemeHandler = (UIColor) -> Void
    var changeThemeHandler: ThemeHandler!

    private let model: Themes = Themes(color1: UIColor.red, color2: UIColor.green, color3: UIColor.blue)

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        view.backgroundColor = navigationController?.navigationBar.barTintColor
    }

    @IBAction func choosedThemeButton(_ sender: UIButton) {
        if sender == themeButtons[0] {
            changeThemeHandler(model.theme1)
            navigationController?.navigationBar.barTintColor = model.theme1
            view.backgroundColor = model.theme1
        } else if sender == themeButtons[1] {
            changeThemeHandler(model.theme2)
            navigationController?.navigationBar.barTintColor = model.theme2
            view.backgroundColor = model.theme2
        } else {
            changeThemeHandler(model.theme3)
            navigationController?.navigationBar.barTintColor = model.theme3
            view.backgroundColor = model.theme3
        }
    }

}
